utility/check_baseline.py

```diff
@@ -0,0 +1,202 @@
+        if len(reduced_ants) < mintotal:
+            observation_good = False
+            verbosef("There are not enough antennas. Only %i antennas avalible."%(len(reduced_ants)),verbose)
+            verbosef("No fewer than %i antennas."%(mintotal),verbose)
+    else:
+        verbosef("NO constraints on the number of antennas",verbose)
+
+    if bins is not None:
+        total = np.histogram((np.sqrt(x**2+y**2)),bins=bins)[0]
+        actual= np.histogram((np.sqrt(x1**2+y1**2)),bins=bins)[0]
+        if not np.all(actual/total >= minfrac):
+            observation_good = False
+        verbosef('# Fraction of available baselines as a function of baseline length:',verbose) 
+        for i in range(len(bins)-1):
+            passfail = "---FAIL---"
+            if (actual/total)[i] >= minfrac[i] :
+                passfail = "    ok     "
+            output_str = "# %4d - %4d m:   %0.3f %% available  >=   %0.3f %% needed -- %s"%(bins[i],bins[i+1],(actual/total)[i],minfrac[i],passfail)
+            verbosef(output_str,verbose)  
+    return observation_good
+
+class projects_contraints(): # I suspect that this is a horrible misuse of a class
+    minfrac = None
+    bins = None
+    mintotal = None
+    
+    
+    
+    
+# Set up standard script options
+parser = optparse.OptionParser(usage='%prog [options] <list of missing antennas>',
+                               description='Checks to see if  the baseline critira for observaitons is met ')
+parser.add_option( "--verbose","-v",action="store_true", default=False,
+                  help="Produce verbose output")
+
+(opts, args) = parser.parse_args()
+missing_ants = args # ['m001','m002','m003']
+projects = {}
+ants  = get_ant_array()
+reduced_ants  =remove_ants(ants,missing_ants,opts.verbose) 
+
+
+# Minimum fraction of available baselines in the above intervals
+# Different projects will have different contraints
+projects['MHONGOOSE'] = projects_contraints()
+projects['MHONGOOSE'].bins =    np.array([0, 50, 100, 200, 400, 1000, 3000, 6000, 9000])
+projects['MHONGOOSE'].minfrac = np.array([0.80, 0.66, 0.66, 0.66, 0.66, 0.66, 0.50, 0.00])
+projects['MHONGOOSE'].mintotal = 58
+
+projects['Fornax'] = projects_contraints()
+projects['Fornax'].bins =    np.array([0, 50, 100, 200, 400, 1000, 3000, 6000, 9000])
+projects['Fornax'].minfrac = np.array([0.80, 0.75, 0.75, 0.75, 0.75, 0.75, 0.50, 0.00])
+projects['Fornax'].mintotal = None
+
+projects['MALS'] = projects_contraints()
+projects['MALS'].bins =    None 
+projects['MALS'].minfrac = None 
+projects['MALS'].mintotal = 56
+
+        
+for key in projects:
+    verbosef('\n',opts.verbose)
+    verbosef(key,opts.verbose)
+    result = test_array(ants,reduced_ants,mintotal=projects[key].mintotal,minfrac=projects[key].minfrac,bins=projects[key].bins,verbose=opts.verbose)
+    passfail = " Can not be observed"
+    if result:
+        passfail = " Meets contraints and may be observed"
+    print('%20s  %s'%(key,passfail))
+    
+ 
```

Comment:
This scripts seems very useful, although I'm not sure why exactly it is being made or when it might be executed. 
 I wonder if it wouldn't be better located in the astrokat repo?  https://github.com/ska-sa/astrokat/tree/master/scripts

Reply:
At the moment this is just quick script for scheduling but it should probably go somewhere useful